File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascal;
        for (int i=1; i<=numRows; i++) {
            vector<int> vect;
            for (int j=0; j<i; j++) {
                vect.push_back(1);
            }
            pascal.push_back(vect);
        }
        for (int i=2; i<numRows; i++) {
            for (int j=1; j<i; j++) {
                pascal[i][j] = pascal[i-1][j] + pascal[i-1][j-1];
            }
        }
        return pascal;
    }
};

File: 118-pascals-triangle/118-pascals-triangle.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut pascal: Vec<Vec<i32>> = Vec::new();
        for i in 1..(num_rows+1) {
            let mut vect: Vec<i32> = Vec::new();
            for j in 0..i {
                vect.push(1);
            }
            pascal.push(vect);
        }
        for i in 1..num_rows as usize {
            for j in 1..i {
                pascal[i][j] = pascal[i-1][j] + pascal[i-1][j-1];
            }
        }
        return pascal;
    }
}
